net-im/skypeforlinux/skypeforlinux-8.56.0.103.ebuild

```diff
@@ -0,0 +1,102 @@
+SLOT="0"
+KEYWORDS="-* ~amd64"
+IUSE="pax_kernel"
+
+S="${WORKDIR}"
+QA_PREBUILT="*"
+RESTRICT="mirror bindist strip" #299368
+
+RDEPEND="
+	app-crypt/libsecret[${MULTILIB_USEDEP}]
+	dev-libs/atk[${MULTILIB_USEDEP}]
+	dev-libs/expat[${MULTILIB_USEDEP}]
+	dev-libs/glib:2[${MULTILIB_USEDEP}]
+	dev-libs/nspr[${MULTILIB_USEDEP}]
+	dev-libs/nss[${MULTILIB_USEDEP}]
+	media-libs/alsa-lib[${MULTILIB_USEDEP}]
+	media-libs/fontconfig:1.0[${MULTILIB_USEDEP}]
+	media-libs/freetype:2[${MULTILIB_USEDEP}]
+	media-libs/libv4l[${MULTILIB_USEDEP}]
+	net-print/cups[${MULTILIB_USEDEP}]
+	sys-apps/dbus[${MULTILIB_USEDEP}]
+	sys-devel/gcc[cxx]
+	virtual/ttf-fonts
+	x11-libs/cairo[${MULTILIB_USEDEP}]
+	x11-libs/gdk-pixbuf:2[${MULTILIB_USEDEP}]
+	x11-libs/gtk+:3[${MULTILIB_USEDEP}]
+	x11-libs/libX11[${MULTILIB_USEDEP}]
+	x11-libs/libXScrnSaver[${MULTILIB_USEDEP}]
+	x11-libs/libXcomposite[${MULTILIB_USEDEP}]
+	x11-libs/libXcursor[${MULTILIB_USEDEP}]
+	x11-libs/libXdamage[${MULTILIB_USEDEP}]
+	x11-libs/libXext[${MULTILIB_USEDEP}]
+	x11-libs/libXfixes[${MULTILIB_USEDEP}]
+	x11-libs/libXi[${MULTILIB_USEDEP}]
+	x11-libs/libXrandr[${MULTILIB_USEDEP}]
+	x11-libs/libXrender[${MULTILIB_USEDEP}]
+	x11-libs/libXtst[${MULTILIB_USEDEP}]
+	x11-libs/libxcb[${MULTILIB_USEDEP}]
+	x11-libs/libxkbfile[${MULTILIB_USEDEP}]
+	x11-libs/pango[${MULTILIB_USEDEP}]"
+
+pkg_setup() {
+	chromium_suid_sandbox_check_kernel_config
+}
+
+src_unpack() {
+	rpm_src_unpack ${A}
+}
+
+src_prepare() {
+	default
+	sed -e "s!^SKYPE_PATH=.*!SKYPE_PATH=${EPREFIX}/opt/skypeforlinux/skypeforlinux!" \
+		-i usr/bin/skypeforlinux || die
+	sed -e "s!^Categories=.*!Categories=Network;InstantMessaging;Telephony;!" \
+		-e "/^OnlyShowIn=/d" \
+		-i usr/share/applications/skypeforlinux.desktop || die
+}
+
+src_install() {
+	dodir /opt
+	cp -a usr/share/skypeforlinux "${D}"/opt || die
+
+	dobin usr/bin/skypeforlinux
+
+	dodoc usr/share/skypeforlinux/*.html
+	dodoc -r usr/share/doc/skypeforlinux/.
+	# symlink required for the "Help->3rd Party Notes" menu entry  (otherwise frozen skype -> xdg-open)
+	dosym ${PF} usr/share/doc/skypeforlinux
+
+	doicon usr/share/pixmaps/skypeforlinux.png
+
+	local res
+	for res in 16 32 256 512; do
+		newicon -s ${res} usr/share/icons/hicolor/${res}x${res}/apps/skypeforlinux.png skypeforlinux.png
+	done
+
+	domenu usr/share/applications/skypeforlinux.desktop
+
+	if use pax_kernel; then
+		pax-mark -m "${ED}"/opt/skypeforlinux/skypeforlinux
+		pax-mark -m "${ED}"/opt/skypeforlinux/resources/app.asar.unpacked/node_modules/slimcore/bin/slimcore.node
+		ewarn "You have set USE=pax_kernel meaning that you intend to run"
+		ewarn "${PN} under a PaX enabled kernel. To do so, we must modify"
+		ewarn "the ${PN} binary itself and this *may* lead to breakage! If"
+		ewarn "you suspect that ${PN} is being broken by this modification,"
+		ewarn "please open a bug."
```

Comment:
We usually pax-mark unconditionally these days. Will it break the program for those not running hardened kernel?

Reply:
Also, 
```
UnderscoreInUseFlag: USE flag 'pax_kernel' uses reserved underscore character
```

Reply:
I can test whether unconditional pax-mark works on my machine.
The same use flag name is used in several packages so I'd expect someone on the security side is handling those. if we decide that keeping the useflag is necessary

Reply:
The "no underscores in USE flags" is relatively new policy, and should be fixed as packages are bumped.
https://projects.gentoo.org/qa/policy-guide/use-flags.html#underscores-in-use-flag-names

Let me know whether it works or not.

Reply:
I've installed -r1 with unconditional pax marking. I joined one video call with several participants, camera, video, and audio worked ok. my mic stopped working after a while, but I'm not sure if that was caused by pax markings, or it was some other issue.
I also noticed that skype no longer blocked keyboard input on my machine, which is a plus.

Reply:
I've had a couple more online meetings with unconditional pax markings and it looks ok. There are occasional hangs and audio glitches, but I'd attribute them to network.